src/ChainRouterGenerateBcLayer.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Component\Routing;
+
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+$refl = new \ReflectionClass(UrlGeneratorInterface::class);
+$generateMethod = $refl->getMethod('generate');
+$methodParameters = $generateMethod->getParameters();
+/** @var \ReflectionParameter $nameParameter */
+$nameParameter = array_shift($methodParameters);
+if ($nameParameter && $nameParameter->hasType() && 'string' === $nameParameter->getType()) {
+    /**
+     * @internal
+     */
+    class ChainRouterGenerateBcLayer
+    {
+        /**
+         * Loops through all registered routers and returns a router if one is found.
+         * It will always return the first route generated.
+         */
+        public function generate(string $name, $parameters = [], $absolute = UrlGeneratorInterface::ABSOLUTE_PATH)
+        {
+            $debug = [];
+
+            foreach ($this->all() as $router) {
+                // if $router does not announce it is capable of handling
+                // non-string routes and $name is not a string, continue
+                if (array_key_exists('_cmf_route', $parameters) && is_object($parameters['_cmf_route']) && !$router instanceof VersatileGeneratorInterface) {
+                    continue;
+                }
+
+                // If $router is versatile and doesn't support this route name, continue
+                if ($router instanceof VersatileGeneratorInterface && !$router->supports($name)) {
```

Comment:
this would be incorrect, we should be using $parameters['_cmf_route']

Reply:
Correct, fixed!